pkg/spoke/hubclientcert/secret_controller.go

```diff
@@ -0,0 +1,75 @@
+package hubclientcert
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	corev1lister "k8s.io/client-go/listers/core/v1"
+)
+
+// hubKubeconfigSecretController watches the HubKubeconfig secret, if the secret is changed, this controller creates/updates the
+// corresponding configuration files from the secret
+type hubKubeconfigSecretController struct {
+	hubKubeconfigDir             string
+	hubKubeconfigSecretNamespace string
+	hubKubeconfigSecretName      string
+	spokeSecretLister            corev1lister.SecretLister
+}
+
+// NewHubKubeconfigSecretController returns a new HubKubeconfigSecretController
+func NewHubKubeconfigSecretController(
+	hubKubeconfigDir, hubKubeconfigSecretNamespace, hubKubeconfigSecretName string,
+	spokeSecretInformer corev1informers.SecretInformer,
+	recorder events.Recorder) factory.Controller {
+	s := &hubKubeconfigSecretController{
+		hubKubeconfigDir:             hubKubeconfigDir,
+		hubKubeconfigSecretNamespace: hubKubeconfigSecretNamespace,
+		hubKubeconfigSecretName:      hubKubeconfigSecretName,
+		spokeSecretLister:            spokeSecretInformer.Lister(),
+	}
+
+	return factory.New().
+		WithInformers(spokeSecretInformer.Informer()).
```

Comment:
we need a queuekey func here so we just enqueue a certain secret

Reply:
done